# ---------------------------
# Create metadata spreadsheet
# ---------------------------

# metadata for all datasets

df_all <- data.frame(
  BiocVersion = "3.15", 
  Genome = NA, 
  SourceType = "FASTQ", 
  SourceVersion = NA, 
  Coordinate_1_based = NA, 
  DataProvider = NA, 
  Maintainer = "Lukas M. Weber <lukas.weber.edu@gmail.com>", 
  RDataClass = "SpatialExperiment", 
  DispatchClass = "Rds", 
  stringsAsFactors = FALSE
)


# metadata for individual datasets

df_Visium_humanDLPFC <- cbind(
  df_all, 
  Title = "Visium_humanDLPFC", 
  Description = paste0(
    "A single sample (sample 151673) of human brain dorsolateral prefrontal ", 
    "cortex (DLPFC) in the human brain, measured using the 10x Genomics Visium ", 
    "platform. This is a subset of the full dataset containing 12 samples from ", 
    "3 neurotypical donors, published by Maynard and Collado-Torres et al. ", 
    "(2021). The full dataset is available from the spatialLIBD Bioconductor ", 
    "package."), 
  SourceUrl = "http://spatial.libd.org/spatialLIBD/", 
  Species = "Homo sapiens", 
  TaxonomyId = "9606", 
  RDataPath = "STexampleData/3_15/Visium_humanDLPFC.rds", 
  stringsAsFactors = FALSE
)

df_Visium_mouseCoronal <- cbind(
  df_all, 
  Title = "Visium_mouseCoronal", 
  Description = paste0(
    "A single coronal section from the mouse brain spanning one hemisphere, ", 
    "measured using the 10x Genomics Visium platform. This dataset was publicly ", 
    "released by 10x Genomics."), 
  SourceUrl = "https://support.10xgenomics.com/spatial-gene-expression/datasets", 
  Species = "Mus musculus", 
  TaxonomyId = "10090", 
  RDataPath = "STexampleData/3_15/Visium_mouseCoronal.rds", 
  stringsAsFactors = FALSE
)

df_seqFISH_mouseEmbryo <- cbind(
  df_all, 
  Title = "seqFISH_mouseEmbryo", 
  Description = paste0(
    "A subset of cells (embryo 1, z-slice 2) from a dataset investigating mouse ", 
    "embryogenesis by Lohoff and Ghazanfar et al. (2020), generated using the ", 
    "seqFISH platform. The full dataset is available from the original ", 
    "publication."), 
  SourceUrl = "https://marionilab.cruk.cam.ac.uk/SpatialMouseAtlas/", 
  Species = "Mus musculus", 
  TaxonomyId = "10090", 
  RDataPath = "STexampleData/3_15/seqFISH_mouseEmbryo.rds", 
  stringsAsFactors = FALSE
)

df_ST_mouseOB <- cbind(
  df_all, 
  Title = "ST_mouseOB", 
  Description = paste0(
    "A single sample from the mouse brain olfactory bulb (OB) measured with ", 
    "the Spatial Transcriptomics platform, published by Stahl et al. (2016). ", 
    "This dataset contains annotations for five cell layers from the original ", 
    "authors."), 
  SourceUrl = "https://www.science.org/doi/10.1126/science.aaf2403", 
  Species = "Mus musculus", 
  TaxonomyId = "10090", 
  RDataPath = "STexampleData/3_15/ST_mouseOB.rds", 
  stringsAsFactors = FALSE
)

df_SlideSeqV2_mouseHPC <- cbind(
  df_all, 
  Title = "SlideSeqV2_mouseHPC", 
  Description = paste0(
    "A single sample of mouse brain from the hippocampus (HPC) and ", 
    "surrounding regions, measured with the Slide-seqV2 platform by Stickels ", 
    "et al. (2020). This dataset contains cell type annotations generated by ", 
    "Cable et al. (2021)."), 
  SourceUrl = "https://www.nature.com/articles/s41587-020-0739-1", 
  Species = "Mus musculus", 
  TaxonomyId = "10090", 
  RDataPath = "STexampleData/3_15/SlideSeqV2_mouseHPC.rds", 
  stringsAsFactors = FALSE
)

df_Chromium_10xJanesick2022_humanBreast <- cbind(
  df_all, 
  Title = "Chromium_10xJanesick2022_humanBreast", 
  Description =  paste0(
    "Chromium data in the 10x Xenium paper - Janesick et al. (2023). 
    High resolution mapping of the breast cancer tumor microenvironment using 
    integrated single cell, spatial and in situ analysis of FFPE tissue. This 
    dataset contains annotations for cell type from the original authors."), 
  SourceUrl = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM7782698",
  Species = "Homo sapiens", 
  TaxonomyId = "9606", 
  RDataPath = "STexampleData/3_18/Chromium_10xJanesick2022_humanBreast.rds",
  stringsAsFactors = FALSE
)


df_Visium_10xJanesick2022_humanBreast <- cbind(
  df_all, 
  Title = "Visium_10xJanesick2022_humanBreast", 
  Description =  paste0(
    "Visium data in the 10x Xenium paper - Janesick et al. (2023). 
    High resolution mapping of the breast cancer tumor microenvironment using 
    integrated single cell, spatial and in situ analysis of FFPE tissue."), 
  SourceUrl = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM7782699", 
  Species = "Homo sapiens", 
  TaxonomyId = "9606", 
  RDataPath = "STexampleData/3_18/Visium_10xJanesick2022_humanBreast.rds",
  stringsAsFactors = FALSE
)

df_Xenium1_10xJanesick2022_humanBreast <- cbind(
  df_all, 
  Title = "Xenium1_10xJanesick2022_humanBreast", 
  Description = paste0(
    "Xenium sample 1 replicate 1 in the 10x Xenium paper - Janesick et al. (2023). 
    High resolution mapping of the breast cancer tumor microenvironment using 
    integrated single cell, spatial and in situ analysis of FFPE tissue."), 
  SourceUrl = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM7780153", 
  Species = "Homo sapiens", 
  TaxonomyId = "9606", 
  RDataPath = "STexampleData/3_18/Xenium1_10xJanesick2022_humanBreast.rds",
  stringsAsFactors = FALSE
)

df_Xenium2_10xJanesick2022_humanBreast <- cbind(
  df_all, 
  Title = "Xenium2_10xJanesick2022_humanBreast", 
  Description =  paste0(
    "Xenium sample 1 replicate 2 in the 10x Xenium paper - Janesick et al. (2023). 
    High resolution mapping of the breast cancer tumor microenvironment using 
    integrated single cell, spatial and in situ analysis of FFPE tissue."), 
  SourceUrl = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM7780154", 
  Species = "Homo sapiens", 
  TaxonomyId = "9606", 
  RDataPath = "STexampleData/3_18/Xenium2_10xJanesick2022_humanBreast.rds",
  stringsAsFactors = FALSE
)

df_Cosmx_humanLung <- cbind(
  df_all, 
  Title = "Cosmx_humanLung", 
  Description =  paste0(
    "A single sample of CosMx human non-small cell lung cancer (NSCLC) data 
    patient 9 sample 1. This dataset was previously released by Nanostring on 
    their website."), 
  SourceUrl = "https://nanostring.com/resources/smi-ffpe-dataset-lung9-rep1-data/", 
  Species = "Homo sapiens", 
  TaxonomyId = "9606", 
  RDataPath = "STexampleData/3_18/Cosmx_humanLung.rds",
  stringsAsFactors = FALSE
)

df_Merscope_humanOvarian <- cbind(
  df_all, 
  Title = "Merscope_humanOvarian", 
  Description =  paste0(
    "A single sample of MERSCOPE human ovarian cancer patient 2 sample 1. This 
    dataset was previously released by Vizgen on their website."), 
  SourceUrl = "https://console.cloud.google.com/storage/browser/vz-ffpe-showcase/HumanOvarianCancerPatient2Slice1;tab=objects?prefix=&forceOnObjectsSortingFiltering=false", 
  Species = "Homo sapiens", 
  TaxonomyId = "9606", 
  RDataPath = "STexampleData/3_18/Merscope_humanOvarian.rds",
  stringsAsFactors = FALSE
)

df_Starmapplus_mouseBrain <- cbind(
  df_all, 
  Title = "Starmapplus_mouseBrain", 
  Description =  paste0(
    "A single sample of the Starmapplus mouse brain well 05 data by Shi, H. 
    et al. (2023). This dataset contains annotations for cell type and tissue 
    regions from the original authors."), 
  SourceUrl = "https://zenodo.org/records/8327576", 
  Species = "Mus musculus", 
  TaxonomyId = "10090", 
  RDataPath = "STexampleData/3_18/Starmapplus_mouseBrain.rds",
  stringsAsFactors = FALSE
)

# combine and save as .csv spreadsheet file

df_combined <- rbind(
  df_Visium_humanDLPFC, 
  df_Visium_mouseCoronal, 
  df_seqFISH_mouseEmbryo, 
  df_ST_mouseOB, 
  df_SlideSeqV2_mouseHPC,
  df_Chromium_10xJanesick2022_humanBreast,
  df_Visium_10xJanesick2022_humanBreast,
  df_Xenium1_10xJanesick2022_humanBreast,
  df_Xenium2_10xJanesick2022_humanBreast,
  df_Cosmx_humanLung,
  df_Merscope_humanOvarian,
  df_Starmapplus_mouseBrain
)

write.csv(df_combined, file = "../extdata/metadata.csv", row.names = FALSE)

